test/e2e/windows/volumes.go

```diff
@@ -47,10 +47,11 @@ var _ = SIGDescribe("Windows volume mounts ", func() {
 				Medium: v1.StorageMediumDefault,
 			},
 		}
-
-		hostMapSource = v1.VolumeSource{
+		hostPathDirectoryOrCreate = v1.HostPathDirectoryOrCreate
+		hostMapSource             = v1.VolumeSource{
 			HostPath: &v1.HostPathVolumeSource{
 				Path: hostMapPath,
+				Type: &hostPathDirectoryOrCreate,
```

Comment:
/cc @ddebroy 
for review too.

Reply:
We can have two tests:
 1. Use `hostPathDirectoryOrCreate` to create the directory
 2. Test volume mounts without `hostPathDirectoryOrCreate`, now that the directory has been created.

/cc @jingxu97

Reply:
Maybe, but I think 2 tests will be redundant here. Significance of this test is the read-only aspect, not hostpathtype.

Reply:
Is there a separate test for hostpathtype on windows?

Reply:
I couldn't find a hostpathtype-specific e2e test, but there are unit ones https://github.com/kubernetes/kubernetes/blob/665e76d97623447d13bb3b33b68591a985b49c9d/pkg/volume/hostpath/host_path_test.go#L522. For example, here's another test that just assumes hostpathtype works https://github.com/kubernetes/kubernetes/blob/dc8b92813645b0072622d9feb85f4bdd48268467/test/e2e/common/kubelet_etc_hosts.go#L156

Reply:
Looks like a reasonable test improvement to me. After creating the directory it will fall through and use the host path as normal.

https://github.com/kubernetes/kubernetes/blob/61af419d386dd9aa05ddfd6e192981ae4ab0ffc5/pkg/volume/hostpath/host_path.go

/lgtm